### ISSUE TITLE
Please add site https://www.mvlempyr.com

### ISSUE THREAD

**Issue by Tenome** [dteviot/WebToEpub#1568]
### Provide URL for web page that contains Table of Contents (list of chapters) of a typical story on the site
https://www.mvlempyr.com/novel/my-house-of-horrors

### Did you try using the Default Parser for the site?  If not, why not?
Yes, but the manual scraper isn't grabbing the URLs for some reason despite showing in the preview. Something with the site blocking it?

### What settings did you use?  What didn't work?
* [Table of Contents](https://www.mvlempyr.com/novel/my-house-of-horrors)
* [URL of first chapter](https://www.mvlempyr.com/chapter/755-1)
* #chapter-content
* #span-28-164
* #image-2836-10

**Comment by Leifman35** [comment]
I would really love it if this website was somehow supported by the scrapper. but it just gives you a "No parser found for this URL. Default Parser will be used. Please specify how to obtain content from each web page."

was going to open a 'site request' for it but searched beforehand and saw this was already requested. so i definitely would wanna join in this request

**Comment by Leifman35** [comment]
Totally randomly, as i am quite clueless in 'how to make a parser' for this website (be it via clicking the bottom and following the instructions as the person that made this post originally or just in general) i think i may have done something that might help with making this website supported.

What i did is went on another supported website (novelbin) and specifically here: https://novelbin.com/b/bro-im-not-an-undead/ and after clicking the web2epub icon i clicked "Edit Chapter URL" and basically removed everything except the very first chapter url, and edited that one with the first chapter url from https://www.mvlempyr.com/novel/bro-im-not-an-undead (the site that we want supported) and basically just put 'https://www.mvlempyr.com/chapter/5384-1' into the <a href="https://novelbin.com/b/bro-im-not-an-undead/chapter-1-dead-forever">Chapter 1 Dead Forever</a> instead of https://novelbin.com/b/bro-im-not-an-undead/chapter-1-dead-forever.

It did managed to scrap the chapter from mvlempyr that way, only additionally adding some weird .gif at the end of the page/chapter. ill attach the .epub that it created but this sorta gives me hope that it is possible? maybe if you somehow configure the mvlempyr site paraser to act similarly to the novelbin one, with basically just somehow to make it grab the first/last chapter and what's between them, while also removing that .gif in the end of each chapter? not sure if it makes sense, but to my small clueless brain... it gives hope? lol. well, hopefully this is helpful and i didnt just ramble with nonsense

[test epub.zip](https://github.com/user-attachments/files/18126050/test.epub.zip)

**Comment by ImLJS** [comment]
@Tenome @Leifman35 I sent a PR for this site. Wait until the code is merged, then you can use the extension for the site.

**Comment by Leifman35** [comment]
> @Tenome @Leifman35 I sent a PR for this site. Wait until the code is merged, then you can use the extension for the site.

You sir are saint and a scholar. 
but in seriousness, thank you :) i really appreciate it.

**Comment by ImLJS** [comment]
@Tenome @Leifman35 Test versions for Firefox and Chrome have been uploaded to https://github.com/dteviot/WebToEpub/releases/tag/developer-build. Pick the one suitable for you, follow the "How to install from Source (for people who are not developers)" instructions at https://github.com/dteviot/WebToEpub/tree/ExperimentalTabMode#user-content-how-to-install-from-source-for-people-who-are-not-developers and let me know how it goes.

**Comment by Leifman35** [comment]
> @Tenome @Leifman35 Test versions for Firefox and Chrome have been uploaded to https://github.com/dteviot/WebToEpub/releases/tag/developer-build. Pick the one suitable for you, follow the "How to install from Source (for people who are not developers)" instructions at https://github.com/dteviot/WebToEpub/tree/ExperimentalTabMode#user-content-how-to-install-from-source-for-people-who-are-not-developers and let me know how it goes.

Just tested it and it Works **PERFECTLY**.

Initially i was going to reply that the website works and all but the 'Titles' of the chapters are not being "gotten" per se, as when one clicks the web2epub icon and prior to generating the epub, it does not show the 'titles' of each chapter in the index page, it just says "Chapter 1" "Chapter 2" and onwards. but that's only in that index page. the actual .epub file generated has each chapter with a title :) so its perfect.

THANK YOU SO MUCH.

**Comment by ImLJS** [comment]
@Leifman35 Parser for the site has been updated. Above Instructions still work. You may see [placeholder] when using the extension on the site, but you can ignore that. After packing of epub, chapter titles will be visible on toc.

**Comment by Leifman35** [comment]
> @Leifman35 Parser for the site has been updated. Above Instructions still work. You may see [placeholder] when using the extension on the site, but you can ignore that. After packing of epub, chapter titles will be visible on toc.

actually i just noticed that, as i just grabbed WebToEpub1.0.1.11.Firefox and as u mentioned, now it shows [placeholder] instead of the previous 'Chapter 1' (without titles) up to the latest chapter that is still does in WebToEpub1.0.1.10.Firefox

So isn't it technically a downgrade? why not keep it how it was (and still is if i load WebToEpub1.0.1.10.Firefox instead of 11)?

I reckon seeing a chapter number without the title is better than just a bunch of [placeholder]'s? or is there something i am missing?

**Comment by ImLJS** [comment]
@Leifman35 sorry for disturbing. For this site, it is extremely difficult to get data as well as chapter titles. Another contributor for the extension said we can use placeholder but after packing of epub, proper chapter titles will be visible in toc of the epub. Anyway, sent the fix for the above, now the chapter titles should come properly in extension as well as in TOC. Above instructions still work, use version number 1.0.1.12

**Comment by Leifman35** [comment]
> @Leifman35 sorry for disturbing. For this site, it is extremely difficult to get data as well as chapter titles. Another contributor for the extension said we can use placeholder but after packing of epub, proper chapter titles will be visible in toc of the epub. Anyway, sent the fix for the above, now the chapter titles should come properly in extension as well as in TOC. Above instructions still work, use version number 1.0.1.12

Yep, can confirm the 1.01.1.12 is just perfect. it both fixes the 1.0.1.10 'lack of titles' which just had chapter numbers and the worsening of it in 1.0.1.11 that had the [placeholder] for all of the chapters without even the numbers.

Now it is flawless with both chapter numbers and names :) THANK YOU.

**Comment by dteviot** [comment]
@Leifman35 

To parapharase ImLJS,  using "[placeholder]" means "WebToEpub doesn't have a title for the chapter at the moment, but when the chapters is fetched, WebToEpub will use the title from the chapter's page."

@ImLJS,  thank you for your hard work.

**Comment by dteviot** [comment]
@Leifman35 

Updated version (1.0.2.0) has been submitted to Firefox and Chrome stores.
Firefox version is available now.
Chrome might be available in a few hours (typical) to 21 days
